### PR TITLE
GCI: Refactored consumers.py

### PR DESCRIPTION
**Pull request by c0derlint**
GCI [Task](https://codein.withgoogle.com/dashboard/task-instances/5882688117407744/): Refactor backend codebase of Real Time Collaboration feature

### CODE REVIEW

File: caffe_app/consumers.py [hunk 1]
```diff
@@ -59,27 +59,47 @@ def ws_connect(message):
...
-    if ('randomId' in data):
-        randomId = data['randomId']
+    each_data = {
+        'UpdateParam': ['layerId', 'param', 'value', 'isProp', 'randomId'],
+        'DeleteLayer': ['layerId'],
+        'AddLayer': ['layer', 'layerId', 'prevLayerId', 'nextLayerId'],
+        'AddComment': ['layerId', 'comment']
+    }
+
+    def update_data(required_data, version_id=None):
```
Comment (Ram81) [inline caffe_app/consumers.py]: Can you move this and make it as independent function.

File: caffe_app/consumers.py [hunk 2]
```diff
@@ -59,27 +59,47 @@ def ws_connect(message):
...
 def ws_receive(message):
-    print('message received...')
+    print('Message received...')
     # param initialization
     data = yaml.safe_load(message['text'])
     action = data['action']
 
-    if ('randomId' in data):
-        randomId = data['randomId']
+    each_data = {
```
Comment (Ram81) [inline caffe_app/consumers.py]: Can you rename this to something more specific

File: caffe_app/consumers.py [hunk 3]
```diff
@@ -59,27 +59,47 @@ def ws_connect(message):
...
 
     if ('networkId' in message.channel_session):
         networkId = message.channel_session['networkId']
 
     if (action == 'ExportNet'):
         # async export call
-        framework = data['framework']
-        net = data['net']
-        net_name = data['net_name']
+        framework, net, net_name = data['framework'], data['net'], data['net_name']
```
Comment (Ram81) [inline caffe_app/consumers.py]: Can you revert this to as it was. This seems to be more confusing.

File: caffe_app/consumers.py [hunk 4]
```diff
@@ -90,103 +110,17 @@ def ws_receive(message):
...
-                'userId': user_id,
-                'action': action,
-                'randomId': randomId,
-                'highlightColor': highlight_color,
-                'username': username
-            })
-        })
-    else:
-        # save changes to database to maintain consistency
+        group_data = update_data(['addHighlightTo', 'removeHighlightFrom', 'userId', 'highlightColor', 'username'])[1]
```
Comment (Ram81) [inline caffe_app/consumers.py]: Can you create an array before call and use that array as a parameter to the call.

File: caffe_app/consumers.py [hunk 5]
```diff
@@ -59,27 +59,47 @@ def ws_connect(message):
...
+    each_data = {
+        'UpdateParam': ['layerId', 'param', 'value', 'isProp', 'randomId'],
+        'DeleteLayer': ['layerId'],
+        'AddLayer': ['layer', 'layerId', 'prevLayerId', 'nextLayerId'],
+        'AddComment': ['layerId', 'comment']
+    }
+
+    def update_data(required_data, version_id=None):
+
+        '''Parses data to include only required keys and return the required object'''
```
Comment (Ram81) [inline caffe_app/consumers.py]: Re-write this as
```
'''
    Parses data to include only required keys and return the required
'''
```

**Comment by c0derlint** [comment]
@Ram81 Updated

File: caffe_app/consumers.py [hunk 6]
```diff
@@ -41,6 +41,26 @@ def fetch_network_version(netObj):
     return network_version
 
 
+def update_data(data, required_data, version_id=None):
+    """
```
Comment (Ram81) [inline caffe_app/consumers.py]: Aren't these supposed to be single quotes?

File: caffe_app/consumers.py [hunk 7]
```diff
@@ -59,18 +79,23 @@ def ws_connect(message):
...
 def ws_receive(message):
-    print('message received...')
+    print('Message received...')
     # param initialization
     data = yaml.safe_load(message['text'])
     action = data['action']
 
-    if ('randomId' in data):
-        randomId = data['randomId']
+    required_keys = {
```
Comment (Ram81) [inline caffe_app/consumers.py]: Rename this too `update_params`

File: caffe_app/consumers.py [hunk 8]
```diff
@@ -90,103 +116,17 @@ def ws_receive(message):
...
-                'highlightColor': highlight_color,
-                'username': username
-            })
-        })
-    else:
-        # save changes to database to maintain consistency
+        group_data = update_data(data, required_keys['UpdateHighlight'])[1]
+
+        Group('model-{0}'.format(networkId)).send(group_data)
+    elif action in required_keys:
```
Comment (Ram81) [inline caffe_app/consumers.py]: paranthesis missing, we should follow the same style as in current code

File: caffe_app/consumers.py (hunk 6 above)
Comment (c0derlint) [reply caffe_app/consumers.py]: According to PEP 257 [here](https://www.python.org/dev/peps/pep-0257/), double quotes are to be used. Please check and confirm.

File: caffe_app/consumers.py [hunk 9]
```diff
@@ -90,103 +116,17 @@ def ws_receive(message):
...
-            Group('model-{0}'.format(networkId)).send({
-                'text': json.dumps({
-                    'layerId': updated_data['layerId'],
-                    'comment': updated_data['comment'],
-                    'action': action,
-                    'version_id': 0,
-                    'randomId': randomId
-                })
-            })
+        updated_data, group_data = update_data(data, required_keys[action], 0)
```
Comment (Ram81) [inline caffe_app/consumers.py]: Why always pass `version_id` as `0`?

File: caffe_app/consumers.py (hunk 6 above)
Comment (Ram81) [reply caffe_app/consumers.py]: We use single quotes, can you make the change.

**Pull request merged**